src/Authenticator/HttpBasicAuthenticator.php

```diff
@@ -33,40 +33,25 @@ class HttpBasicAuthenticator extends AbstractAuthenticator implements StatelessI
      * @return \Authentication\Authenticator\ResultInterface
      */
     public function authenticate(ServerRequestInterface $request): ResultInterface
-    {
-        $user = $this->getUser($request);
-
-        if (empty($user)) {
-            return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
-        }
-
-        return new Result($user, Result::SUCCESS);
-    }
-
-    /**
-     * Get a user based on information in the request. Used by cookie-less auth for stateless clients.
-     *
-     * @param \Psr\Http\Message\ServerRequestInterface $request Request object.
-     * @return \ArrayAccess|array|null User entity or null on failure.
-     */
-    public function getUser(ServerRequestInterface $request)
     {
         $server = $request->getServerParams();
-        if (!isset($server['PHP_AUTH_USER']) || !isset($server['PHP_AUTH_PW'])) {
-            return null;
-        }
+        $username = $server['PHP_AUTH_USER'] ?? '';
+        $password = $server['PHP_AUTH_PW'] ?? '';
 
-        $username = $server['PHP_AUTH_USER'];
-        $password = $server['PHP_AUTH_PW'];
-
-        if (!is_string($username) || $username === '' || !is_string($password) || $password === '') {
-            return null;
+        if ($username === '' || $password === '') {
```

Comment:
Could we have non-scalar values in username or password here? I think we should be returning credentials missing if either password or username are not scalar.

Reply:
I am unaware how user could provide non string values for these `PHP_AUTH_USER` and `PHP_AUTH_PW` env vars.

Reply:
True, I've been burned too many times from PHP surprises I guess. :smile: